manifests/params.pp

```diff
@@ -1,120 +1,26 @@
 # @summary This class provides a number of parameters.
 # @api private
 class redis::params {
```

Comment:
This class should go away completely and be replaced with hiera data in module.

Reply:
My motivation to keep it was that this was much easier to understand. I also wanted to limit the changes in this PR to making sure the data types are correct and the REFERENCE.md is useful.

Reply:
There's also more than one class that inherits from params.pp
Getting rid of params.pp in puppet/prometheus left something of a mess IMO.  See https://github.com/voxpupuli/puppet-prometheus/issues/184